### PR TITLE
Curator notes

### PR DESCRIPTION
**Pull request by daisieh**
Curators can add notes in the dryad.curatorNote metadata field and they will display at the top of the Edit Metadata page and any view item: show full metadata pages. These will only be visible to curators and admins, not to users.

### CODE REVIEW

File: dspace/config/crosswalks/DIM2DATACITE.xsl [hunk 1]
```diff
@@ -322,7 +322,7 @@
 				<descriptions>
 					<description descriptionType="Other">
 					        <!-- Exclude provenance fields, since these often contain private information -->
-						<xsl:value-of select="dspace:field[@element='description' and not(@qualifier='provenance')]"/>
+						<xsl:value-of select="dspace:field[@element='description']"/>
```
Comment (ryscher) [inline dspace/config/crosswalks/DIM2DATACITE.xsl]: Why this change? We don't want to send provenance metadata (such as rejection messages) to DataCite.

Comment (daisieh) [reply dspace/config/crosswalks/DIM2DATACITE.xsl]: It doesn't; everything in the metadata.hide area of the config is not sent to DataCite. That was the correct way to do it, not this other way.

Comment (daisieh) [reply dspace/config/crosswalks/DIM2DATACITE.xsl]: If that isn't the case (I thought I'd tested it adequately), we can roll this one change back out, but this is a band-aid on a bigger bug if so.

Comment (ryscher) [reply dspace/config/crosswalks/DIM2DATACITE.xsl]: I will test -- The documentation in the config indicates that the "hide" settings only apply to the DSpace GUI, not to the DataCite process. And there may be other command-line tools that use the XSL transform and send content to DataCite without running through the full DSpace process.

Comment (daisieh) [reply dspace/config/crosswalks/DIM2DATACITE.xsl]: Okay. It is easy enough for me to roll out that commit from the crosswalk.

**Comment by ryscher** [comment]
This looks ok, and doesn't seem to hurt anything. So it can be merged. 

But I'm not sure the private notes are displaying everywhere. Places I expected the private notes, but they don't show:
- Edit metadata page, while item is in curation
- Item view page, both while in curation and after archiving

**Comment by daisieh** [comment]
I'm going to merge this and then test more on production.

**Comment by daisieh** [comment]
It looks fine on production. I see notes on Edit Metadata in curation and in archive, and on the Show Full Metadata pages for View Item.
